src/app/forms/custom-sections/adr-tank-statement-un-number-edit/adr-tank-statement-un-number-edit.component.html

```diff
@@ -1,40 +1,44 @@
-<ng-container *ngIf="control && formArray.controls[0]">
-  <app-field-error-message
-    [error]="!formArray.controls[0].value && control.invalid && (control.touched || submitted) ? control.meta.customErrorMessage : ''"
-  ></app-field-error-message>
+<ng-container *ngIf="control && formArray.controls.length">
+  <app-field-error-message [error]="control.touched ? control.getError('custom')?.message : ''"></app-field-error-message>
 
-  <ng-container *ngIf="formArray.controls.length === 1">
+  <ng-container *ngIf="formArray.controls.length === 1; else multiple">
     <app-text-input
-      [id]="control.meta.name"
-      [name]="formArray.controls[0].meta.name"
-      [label]="formArray.controls[0].meta.label + ' 1'"
-      [customId]="formArray.controls[0].meta.customId"
+      [id]="'UN_number_1'"
```

Comment:
pure curiosity, why didn't this work using the meta stuff?

Reply:
oh I see, ignore

Reply:
Ah, long story, this is a form array where the controls can be added or removed, theoretically out of order. The meta property is passed when constructing the custom form control, so when you add/remove a control you'd need to manually update all the controls labels so it has the correct number (to make clicking on global errors go to the correct text input). Doing it in html make it so that this happens automatically - e.g. adding a control causes a re-render so all the input bindings update properly.